### ISSUE TITLE
nanosleep() implicitly converts floats to zeroes

### ISSUE THREAD

**Issue by socketpair** [luaposix/luaposix#310]
i.e. `nanosleep(2.1, 10.1)` is be executed as `nanosleep(0, 0)`

**Comment by socketpair** [comment]
ping

**Comment by tst2005** [comment]
[nanosleep](http://luaposix.github.io/luaposix/modules/posix.time.html#nanosleep) needs a  [PosixTimespec](http://luaposix.github.io/luaposix/modules/posix.time.html#PosixTimespec) argument  that seems to be defined with only integer values.

You can make a wrapper to fix your wrong values
```lua
local function PosixTimespecFix(t)
  t.tv_sec=math.floor(t.tv_sec)
  t.tv_nsec=math.floor(t.tv_nsec)
  return t
end
```
and use
```lua
nanosleep(PosixTimespecFix({tv_sec=2.1, tv_nsec=10.1}))
```

**Comment by socketpair** [comment]
yes, I understand and have done exactly this workaround. But problem still exists.

**Comment by socketpair** [comment]
The problem in the handling of numbers inside the binding.
